devito/operator/operator.py

```diff
@@ -493,6 +493,9 @@ def _lower_iet(cls, uiet, profiler=None, **kwargs):
         # Target-independent optimizations
         minimize_symbols(graph)
 
+        # If necessary, sort frees into a specific order
+        sort_frees(graph)
```

Comment:
Imho this should enforced at construction time (inside `definitions.py`?), not through postprocessing like here

Reply:
I've moved the sort to `definitions.py`. It uses the new `_C_free_priority` property added to `LocalObject` types, as we briefly discussed...what do you think?...